### PR TITLE
[diem framework] Add metadata field to preburn requests

### PR DESCRIPTION
**Pull request by tzakian**
This adds an opaque-to-Move `metadata` field to each preburn request in the preburn queue. This is currently not used or set, but put in for future use as it's easier to do so now and avoid a data upgrade later on.

This PR also updates the json-rpc view for preburn queues to support returning the metadata associated with each preburn request in the preburn queue. 

Due to the changes to json-rpc, we should probably land this after #7973

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

File: json-rpc/docs/type_preburn_queue.md [hunk 1]
```diff
@@ -2,7 +2,15 @@
...
-| ---------- | -------------------------------- | -------------------------------------------------------      |
-| currency   | string                           | The currency that all preburn requests in this queue are in. |
-| preburns   | List<[Amount](type_amount.md)>   | A list of preburn requests. All in the same currency.        |
+| Name       | Type                                                   | Description                                                  |
+| ---------- | --------------------------------                       | -------------------------------------------------------      |
+| currency   | string                                                 | The currency that all preburn requests in this queue are in. |
+| preburns   | List<[PreburnWithMetadata](#type-preburnwithmetadata)> | A list of preburn requests. All in the same currency.        |
+
+
+## PreburnWithMetadata - type
```
Comment (xli) [inline json-rpc/docs/type_preburn_queue.md:11]: for above "#type-preburnwithmetadata" to work, you need change this header to "Type PreburnWithMetadata"

**Review by xli — APPROVED** [review]
LGTM for the JSON-RPC related change, approve for that.

**Comment by moeziniaf** [comment]
can we get the metadata sent with BurnEvents as well?

**Pull request closed without merging**